src/shapes/ActiveSelection.ts

```diff
@@ -76,22 +76,29 @@ export class ActiveSelection extends Group {
   }
 
   /**
+   * Change an object so that it can be part of an active selection.
+   * this method is called by multiselectAdd from canvas code.
    * @private
    * @param {FabricObject} object
    * @param {boolean} [removeParentTransform] true if object is in canvas coordinate plane
-   * @returns {boolean} true if object entered group
    */
   enterGroup(object: FabricObject, removeParentTransform?: boolean) {
-    if (object.group) {
-      //  save ref to group for later in order to return to it
-      const parent = object.group;
-      parent._exitGroup(object);
-      // make sure we are setting the correct owning group
-      // in case `object` is transferred between active selections
-      !(parent instanceof ActiveSelection) && (object.__owningGroup = parent);
+    // This condition check that the object has currently a group, and the group
+    // is also its parent, meaning that is not in an active selection, but is
+    // in a normal group.
+    if (object.parent && object.parent === object.group) {
+      // Disconnect the object from the group functionalities, but keep the ref parent intact
+      // for later re-enter
+      object.parent._exitGroup(object);
+      // in this case the object is probably inside an active selection.
+    } else if (object.group && object.parent !== object.group) {
+      // in this case group.remove will also clear the old parent reference.
+      object.group.remove(object);
```

Comment:
please check this comment here @ShaMan123 .
I do not know if the deletion of object.parent in this case is wanted.
On line 92 that doesn't happen, and i don't understand why this should be different.

Reply:
object.group === ActiveSelection instance so remove will not remove the parent ref
I agree instaof ActiveSeletion will make this code eaasier